NDTensors/src/tensor/tensor.jl

```diff
@@ -33,7 +33,8 @@ end
 function Tensor{ElT,N,StoreT,IndsT}(
   ::NeverAlias, storage::TensorStorage, inds
 ) where {ElT,N,StoreT<:TensorStorage,IndsT}
-  return Tensor{ElT,N,StoreT,IndsT}(AllowAlias(), copy(storage), inds)
+  store = copy(storage)
+  return Tensor{ElT,N,typeof(store),IndsT}(AllowAlias(), store, inds)
 end
```

Comment:
I don't think this proposed change is a good idea. Calling `Tensor{ElT,N,StoreT,IndsT}(NeverAlias(), storage, inds)` should only ever construct an object of type `Tensor{ElT,N,StoreT,IndsT}`. More generally, constructors of a type `T` should obey `T(args...) isa T`, as we have discussed in the past. We should consider that to be a "law" of designing Julia code, even if it isn't enforced by the compiler.

Reply:
Ah right, the type conversion did seem sketchy to me. I'll rework this. Thanks!